ServerCore/Pages/Puzzles/Feedback.cshtml.cs

```diff
@@ -16,27 +17,30 @@ public class FeedbackModel : EventSpecificPageModel
     {
         private readonly PuzzleServerContext _context;
 
-        public FeedbackModel(PuzzleServerContext context)
+        public FeedbackModel(PuzzleServerContext context, UserManager<IdentityUser> userManager) : base(context, userManager)
         {
             _context = context;
         }
 
         public IList<Feedback> Feedbacks { get; set; }
         public Puzzle Puzzle { get; set; }
+        public bool IsCurrentUserPuzzleAuthor { get; set; }
 
         /// <summary>
         /// Gets the feedback and puzzle associated with the given ID
         /// </summary>
         public async Task<IActionResult> OnGetAsync(int id)
         {
-            Feedbacks = await _context.Feedback.Where((f) => f.Puzzle.ID == id).ToListAsync();
+            Feedbacks = await _context.Feedback.Where((f) => f.Puzzle.ID == id).Include("Submitter").ToListAsync();
             Puzzle = await _context.Puzzles.Where((p) => p.ID == id).FirstOrDefaultAsync();
 
             if (Puzzle == null) 
             { 
                 return NotFound(); 
             } 
 
+            IsCurrentUserPuzzleAuthor = (await _context.PuzzleAuthors.Where((a) => a.Puzzle == Puzzle && a.Author == LoggedInUser).FirstOrDefaultAsync() != null);
```

Comment:
If you're just using this for auth, you should be able to do it with attributes. If it serves another purpose, OK to keep.